cmd/capability/cmd.go

```diff
@@ -0,0 +1,69 @@
+package capability
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// All the base paths for OCM resources used for capabilities
+	basePath         = "/api/accounts_mgmt/v1"
+	labelsPath       = "/labels"
+	organizationPath = basePath + "/organizations"
+	subscriptionPath = basePath + "/subscriptions"
+
+	hibernation          = "hibernation"
+	autoscaling          = "autoscaling"
+	upgradeChannelChange = "upgradeChannelChange"
+	ovn                  = "ovn"
+
+	// Source for all capabilitiy keys. Add new keys here when adding a capability to the command
+	hibernationKey          = "capability.organization.hibernate_cluster"
+	autoscalingKey          = "capability.cluster.autoscale_clusters"
+	upgradeChannelChangeKey = "capability.organization.allow_set_upgrade_channel_group"
+	ovnKey                  = "capability.organization.ovn_cluster"
+)
+
+type CapabilityBody struct {
+	Internal     bool   `json:"internal"`
+	Key          string `json:"key"`
+	ResourceType string `json:"type"`
+	Value        string `json:"value"`
+}
+
+func NewCmdCapability() *cobra.Command {
+	var capabilityCmd = &cobra.Command{
+		Use:   "capability",
+		Short: "Manage capabilites for OCM Organizations",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cmd.Help()
+			if err != nil {
+				fmt.Println("Error calling cmd.Help(): ", err.Error())
+				return
+			}
+		},
+	}
+
+	// Add subcommands
+	capabilityCmd.AddCommand(newAddCmd())
+	capabilityCmd.AddCommand(newRemoveCmd())
+
+	return capabilityCmd
+}
+
+func getCapabilityKey(capability string) string {
+
+	switch capability {
```

Comment:
This caught me up at first, thinking these were somehow magic strings.  Would this be easier to maintain if it is depicted as a map[string]string and checking if a key exists?

Reply:
This is indeed basically a fake map. I tried making a var of type map[string]string at the package level, however Go doesn't allow this. In order to create a map that could be used, we'd need a function that builds the map and returns it to the caller, then the caller would need to attempt to get the value out with the key. This is a shortcut that recreates the functionality of a map, using constant string values to make sure there's no misspellings. I'll leave a comment explaining this